src/nspawn/nspawn-network.c

```diff
@@ -168,12 +168,31 @@ static int add_veth(
         return 0;
 }
 
+static char *shorten_ifname(char *ifname) {
+        unsigned char hash = 0xff;
+
+        assert(ifname);
+
+        if (strlen(ifname) >= IFNAMSIZ) {
+
+                /* Create a simple 8-bit hash of an interface name */
+                for (char *pos = ifname; *pos != 0; pos++)
+                        hash = ((hash << 1) | (hash >> 7)) ^ *pos;
```

Comment:
please don't make up hash functions, please use the existing siphash implementation we already have.

Reply:
I only discovered that later, I'll look into it.